src/wasm/activity-service/service/service.go

```diff
@@ -358,7 +359,12 @@ func (s *service) combineActivity(activities []activity.Activity, manufacturer t
 			newActivity.Sessions = append(newActivity.Sessions, cur.Sessions[1:]...)
 		}
 
-		newActivity.UnrelatedMessages = append(newActivity.UnrelatedMessages, cur.UnrelatedMessages...)
+		for j := 0; j < len(cur.UnrelatedMessages); j++ {
+			if cur.UnrelatedMessages[j].Num == mesgnum.SplitSummary {
+				continue // TODO: Still failed to upload to Garmin Connect if we include this message.
```

Comment:
Is this still TODO or will fix the current issue?

Reply:
It's a workaround, I placed TODO there to mark that this need to be addressed later in the future when we know how to handle split summary message correctly.